aiperf/common/models/record_models.py

```diff
@@ -240,6 +240,20 @@ class RequestRecord(AIPerfBaseModel):
         "This can be used to trace internal latency in order to identify bottlenecks or other issues.",
         ge=0,
     )
+    was_cancelled: bool = Field(
+        default=False,
+        description="Whether the request was cancelled during execution.",
+    )
+    cancel_after_ns: int = Field(
+        default=0,
+        ge=0,
+        description="The delay in nanoseconds after which the request should be cancelled, as specified in the credit drop message.",
+    )
+    cancellation_time_ns: int | None = Field(
```

Comment:
I typically have been using `perf_ns` in names of things that use `perf_counter_ns` to make it more clear to the developer. So probably would name this as `cancellation_perf_ns`